packages/flutter_test/test/controller_test.dart

```diff
@@ -1015,6 +1015,468 @@ void main() {
+          await tester.pump();
+        }
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollLeftAction: false, hasScrollRightAction: true),
+          reason: 'When fully scrolled, a scrollview should only support right scrolls.',
+        );
+
+        tester.semantics.scrollRight();
+        await tester.pump();
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollLeftAction: true, hasScrollRightAction: true),
+          reason: 'When partially scrolled, a scrollview should be able to support both left and right scrolls.',
+        );
+      });
+
+      testWidgets('scrollUp and scrollDown scrolls up and down respectively', (WidgetTester tester) async {
+        await tester.pumpWidget(MaterialApp(
+          home: ListView(
+            children: <Widget>[
+              SizedBox(
+                height: tester.binding.window.physicalSize.height * 1.5,
+                width: 40,
+              )
+            ],
+          ),
+        ));
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollUpAction: true, hasScrollDownAction: false),
+          reason: 'When not yet scrolled, a scrollview should only be able to support left scrolls.',
+        );
+
+        tester.semantics.scrollUp();
+        await tester.pump();
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollUpAction: true, hasScrollDownAction: true),
+          reason: 'When partially scrolled, a scrollview should be able to support both left and right scrolls.',
+        );
+
+        // This will scroll the listview until it's completely scrolled to the right.
+        final SemanticsFinder upScrollable = find.semantics.byAction(SemanticsAction.scrollUp);
+        while (upScrollable.tryEvaluate()) {
+          tester.semantics.scrollUp(scrollable: upScrollable);
+          await tester.pump();
+        }
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollUpAction: false, hasScrollDownAction: true),
+          reason: 'When fully scrolled, a scrollview should only support right scrolls.',
+        );
+
+        tester.semantics.scrollDown();
+        await tester.pump();
+
+        expect(
+          find.semantics.scrollable(),
+          containsSemantics(hasScrollUpAction: true, hasScrollDownAction: true),
+          reason: 'When partially scrolled, a scrollview should be able to support both left and right scrolls.',
+        );
+      });
+
+      testWidgets('increase causes semantic increase', (WidgetTester tester) async {
+        bool invoked = false;
+        await tester.pumpWidget(MaterialApp(
+            home: Material(
+              child: _StatefulSlider(
+                initialValue: 0,
+                onChanged: (double _) {invoked = true;},
+              ),
+            )
```

Comment:
very nitty: this looks like it is indented a little to far

Reply:
Fixed.